qemu/tests/virtio_serial_throttling.py

```diff
@@ -0,0 +1,53 @@
+from avocado.utils import process
+from virttest import error_context
+
+from qemu.tests.virtio_serial_file_transfer import transfer_data
+
+import socket
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Induce throttling by opening chardev but not reading data from it:
+    1) Start guest with virtio-serial with unix option
+    2) On the host, open the chardev but don't read from it
+    3) In the guest, write to the virtio-serial port.
+    4) repeat step 1 to 3 with tcp option.
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def get_chardev_params(vm, serial_id):
+        serial_device = vm.devices.get(serial_id)
+        chardev_id = serial_device.get_param('chardev')
+        chardev_device = vm.devices.get(chardev_id)
+        chardev_params = chardev_device.params
+        return chardev_params
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    serial_id = params.objects('serials')[-1]
+    chardev_path = get_chardev_params(vm, serial_id)['path']
+    session = vm.wait_for_login()
+    sock = socket.socket(socket.AF_UNIX)
+    sock.connect(chardev_path)
+    a = session.cmd_output('dd if=/dev/zero of=/dev/vport0p2')
```

Comment:
The output port is not correct, please use `/dev/virtio-ports/*serial_id*`.
@nanliu-r `/dev/virtio-ports/*serial_id*`  is a soft link of `/dev/vport0p1`, but i have no idea with the name format of `vport0p1`. so suggest use `/dev/virtio-ports/*serial_id*` , please help review, thanks!

Reply:
I think they are the same, will check this using guest.

Reply:
> The output port is not correct, please use `/dev/virtio-ports/*serial_id*`.
> @nanliu-r `/dev/virtio-ports/*serial_id*` is a soft link of `/dev/vport0p1`, but i have no idea with the name format of `vport0p1`. so suggest use `/dev/virtio-ports/*serial_id*` , please help review, thanks!

They are the same, but the second foramt is  `/dev/virtio-ports/*serial_name*`.

Reply:
@nanliu-r  You are right, it is `serial name` not `serial id`.
My key point is, we can not use `/dev/vport0p1` directly, it is not related with any qemu option.
We should use `/dev/virtio-ports/*serial_name*` which is related with our setting.
If needed, get `/dev/vport0p1` from `/dev/virtio-ports/*serial_name*`, instead of use it directly.

Reply:
> @nanliu-r You are right, it is `serial name` not `serial id`.
> My key point is, we can not use `/dev/vport0p1` directly, it is not related with any qemu option.
> We should use `/dev/virtio-ports/*serial_name*` which is related with our setting.
> If needed, get `/dev/vport0p1` from `/dev/virtio-ports/*serial_name*`, instead of use it directly.

ok, it makes sense and clearer, agree with you.